### PR TITLE
Secure Socket Funneling 1.1.0

### PR DESCRIPTION
**Pull request by dgarske**

### CODE REVIEW

**Review by MJSPollard — DISMISSED** [review]
All SSF unit tests passed when testing with https://github.com/wolfSSL/wolfssl/pull/1739.
I opt for this to be merged into master.

File: ssf-1.1.0/README_WOLF.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    $ `git clone https://github.com/wolfSSL/wolfssl.git`
+    OR
+    $ `scp wolfssl-3.15.3.tar.gz ubuntu@204.48.22.68:~/ssfwolfbuild`
+    $ `tar -xzf wolfssl-3.15.3.tar.gz`
+    $ `mv wolfssl-3.15.3 wolfssl`
+
+2. Execute the following commands from the wolfSSL root directory
+
+    $ `cd wolfssl`
```
Comment (ejohnstown) [inline ssf-1.1.0/README_WOLF.md:23]: Above it says to start in the wolfSSL root directory. Do we want to change directory into the headers directory here?

File: ssf-1.1.0/README_WOLF.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+   To Enable Intel Optimizations use:
+   $ `./configure --enable-asio --enable-sp --enable-intelasm --enable-aesni --enable-intelrand`
+
+3. Download SSF 1.1.0 and extract
+
+    https://github.com/securesocketfunneling/ssf/releases
+    $ `wget https://github.com/securesocketfunneling/ssf/archive/1.1.0.tar.gz`
+    $ `tar -xzf 1.1.0.tar.gz`
+    Extracts to ssf-1.1.0
```
Comment (ejohnstown) [inline ssf-1.1.0/README_WOLF.md:39]: Where are we downloading and extracting this to? Into the wolfSSL root directory? It's own directory parallel to wolfSSL? Somewhere else?

File: ssf-1.1.0/README_WOLF.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+4. Download ASIO 1.65.0 WolfSSL modified
+
+    $ `scp asio_1_65_1.tar.gz ubuntu@204.48.22.68:~/ssfwolfbuild`
+    $ `tar -xzf asio_1_65_1.tar.gz`
+
+5. Download boost 1_65_1 into third_party/boost
```
Comment (ejohnstown) [inline ssf-1.1.0/README_WOLF.md:46]: Download where?

File: ssf-1.1.0/README_WOLF.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+    https://www.boost.org/users/history
+    $ `wget https://dl.bintray.com/boostorg/release/1.65.1/source/boost_1_65_1.tar.gz`
```
Comment (ejohnstown) [inline ssf-1.1.0/README_WOLF.md:49]: URL has changed. It is now found at https://boostorg.jfrog.io/artifactory/main/release/1.65.1/source/boost_1_65_1.tar.gz

File: ssf-1.1.0/README_WOLF.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+3. Download SSF 1.1.0 and extract
+
+    https://github.com/securesocketfunneling/ssf/releases
+    $ `wget https://github.com/securesocketfunneling/ssf/archive/1.1.0.tar.gz`
+    $ `tar -xzf 1.1.0.tar.gz`
+    Extracts to ssf-1.1.0
+
+4. Download ASIO 1.65.0 WolfSSL modified
+
+    $ `scp asio_1_65_1.tar.gz ubuntu@204.48.22.68:~/ssfwolfbuild`
```
Comment (ejohnstown) [inline ssf-1.1.0/README_WOLF.md:43]: When do we make ssfwolfbuild?

File: ssf-1.1.0/README_WOLF.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    $ `tar -xzf asio_1_65_1.tar.gz`
+
+5. Download boost 1_65_1 into third_party/boost
+
+    https://www.boost.org/users/history
+    $ `wget https://dl.bintray.com/boostorg/release/1.65.1/source/boost_1_65_1.tar.gz`
+    $ `tar -xzf boost_1_65_1.tar.gz`
+    $ `cd boost_1_65_1`
+    
+    Link boost to SSF from boost_1_65_1 directory
```
Comment (ejohnstown) [inline ssf-1.1.0/README_WOLF.md:53]: I had to guess where to do this. I did it from the ssf directory I got unarchiving the ssf sources above.

File: ssf-1.1.0/README_WOLF.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    $ `mv ./boost/asio ./boost/asio_old`
+    $ `ln -s ../../../osp/asio_1_65_1 ./boost/asio`
+
+    Link boost to SSF from boost_1_65_1 directory
+    $ `cd ../ssf-1.1.0`
```
Comment (ejohnstown) [inline ssf-1.1.0/README_WOLF.md:58]: Again which directory? Where I installed everything, boost_1_65_1 wasn't at ../..

File: ssf-1.1.0/README_WOLF.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+    $ `ln -s ../../boost_1_65_1 ./third_party/boost`
+
+
+## Building
+
+Execute the following commands from the SSF root directory
+
+    $ `mkdir build`
+    $ `cd build`
+    $ `cmake -DCMAKE_BUILD_TYPE=Release -DWOLFSSL="/usr/local" ../`
```
Comment (ejohnstown) [inline ssf-1.1.0/README_WOLF.md:69]:
```
CMake Error at cmake/external/ExternalPackageHelpers.cmake:246 (message):
  [Boost] # Cannot determine Boost's source directory (NOTFOUND)
Call Stack (most recent call first):
  cmake/external/ExternalPackageHelpers.cmake:818 (external_error)
  cmake/external/ExternalPackageHelpers.cmake:929 (external_search_source_path)
  cmake/external/boost/FindBoost.cmake:74 (externals_build)
  cmake/external/boost/FindBoost.cmake:256 (boost_build_components)
  CMakeLists.txt:193 (find_package)
```

**Review by ejohnstown — CHANGES_REQUESTED** [review]

**Comment by dgarske** [comment]
Note: GitHub is not letting either Jacob or I reopen this PR so I'll just open a new one with a different branch name

**Comment by dgarske** [comment]
My fork was not tied upstream. Setup new proper fork. Attempting to re-open again based on that. Would like to use this PR to avoid loosing the comments.

**Pull request closed without merging**